library/L1_Peripheral/lpc40xx/capture.hpp

```diff
@@ -0,0 +1,195 @@
+        inline static const Timer::ChannelPartial_t kTimerPartial1 = {
+        .timer_register = LPC_TIM1,
+        .power_id       = SystemController::Peripherals::kTimer1,
+        .irq            = IRQn::TIMER1_IRQn,
+        .user_callback  = &timer1_isr,
+        };
+
+        inline static IsrPointer timer2_isr                 = nullptr;
+        inline static const Timer::ChannelPartial_t kTimerPartial2 = {
+        .timer_register = LPC_TIM2,
+        .power_id       = SystemController::Peripherals::kTimer2,
+        .irq            = IRQn::TIMER2_IRQn,
+        .user_callback  = &timer2_isr,
+        };
+
+        inline static IsrPointer timer3_isr                 = nullptr;
+        inline static const Timer::ChannelPartial_t kTimerPartial3 = {
+        .timer_register = LPC_TIM3,
+        .power_id       = SystemController::Peripherals::kTimer3,
+        .irq            = IRQn::TIMER3_IRQn,
+        .user_callback  = &timer3_isr,
+        };
+
+     public:
+        inline static const Timer::Channel_t kCaptureTimer0 =
+        {
+            .channel = kTimerPartial0,
+            .isr = TimerHandler<kTimerPartial0>
+        };
+        inline static const Timer::Channel_t kCaptureTimer1 =
+        {
+            .channel = kTimerPartial1,
+            .isr = TimerHandler<kTimerPartial1>
+        };
+        inline static const Timer::Channel_t kCaptureTimer2 =
+        {
+            .channel = kTimerPartial2,
+            .isr = TimerHandler<kTimerPartial2>
+        };
+        inline static const Timer::Channel_t kCaptureTimer3 =
+        {
+            .channel = kTimerPartial3,
+            .isr = TimerHandler<kTimerPartial3>
+        };
+    };  // struct Channel
+
+
+    static void TimerHandler(const Timer::ChannelPartial_t & channel)
+    {
+        if (*channel.user_callback != nullptr)
+        {
+            (*channel.user_callback)();
+        }
+        channel.timer_register->IR |= 0x30;
+    }
+
+    typedef struct
+    {
+        uint32_t count;
+        uint32_t flags;
+    } CaptureStatus_t;
+
+    static constexpr sjsu::cortex::InterruptController kInterruptController =
+        sjsu::cortex::InterruptController();
+
+    explicit constexpr Capture(
+        const Timer::Channel_t &timer,
+        const sjsu::SystemController &system_controller =
+            DefaultSystemController(),
+        const sjsu::InterruptController &interrupt_controller =
+            kInterruptController
+        )
+        : timer_(timer),
+          system_controller_(system_controller),
+          interrupt_controller_(interrupt_controller)
+    {
+    };
+
+    Status Initialize(
+        units::frequency::hertz_t frequency,
+        IsrPointer isr = nullptr, int32_t priority = -1) const override
+    {
+        // NOTE: Same initialization as Timer library's Initialize()
+        system_controller_.PowerUpPeripheral(timer_.channel.power_id);
+        SJ2_ASSERT_FATAL(
+            frequency.to<uint32_t>() != 0,
```

Comment:
```suggestion
            frequency != 0_Hz,
```
You don't need to convert. You can compare the the Hz literal.